### PR TITLE
scan: patch request hook

### LINKED ISSUE #676: Implement hook to get request information
Make sure it has all data necessary for scan to operate. If it does not then some changes on chain may be required.

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #676 

[Document](https://docs.google.com/spreadsheets/d/1JwzzcJl0bDQdi1I8T3KiZloRX6RaGvZ3Vi4ZIRmUbCM/edit#gid=469360780)

[API](http://d3n-debug.bandprotocol.com:1317/zoracle/request/1)

### CODE REVIEW

File: scan/src/rpc/RequestHook.re [hunk 1]
```diff
@@ -17,47 +17,75 @@ module Report = {
...
-    codeHash: Hash.t,
-    params: array((string, Js.Json.t)),
-    targetBlock: int,
+    requestID: int,
+    oracleScriptID: int,
+    calldata: JsBuffer.t,
+    requestedValidators: list(Address.t),
+    sufficientValidatorCount: int,
+    expirationHeight: int,
+    isResolved: bool,
```
Comment (taobun) [inline scan/src/rpc/RequestHook.re]: isResolved changed to resolveStatus already.

File: scan/src/rpc/RequestHook.re [hunk 2]
```diff
@@ -17,47 +17,75 @@ module Report = {
...
-      reportedAtTime: json |> at(["reportedAtTime"], moment),
-      values: json |> at(["value"], dict(x => x)) |> Js.Dict.entries,
+      reporter: json |> field("reporter", string) |> Address.fromBech32,
+      txHash: json |> at(["tx", "hash"], string) |> Hash.fromHex,
+      reportedAtHeight: json |> at(["tx", "height"], intstr),
+      reportedAtTime: json |> at(["tx", "timestamp"], moment),
+      values: json |> field("value", list(RawDataReport.decode)),
     };
 
   let decodeReports = json => JsonUtils.Decode.(json |> field("reports", list(decode)));
```
Comment (taobun) [inline scan/src/rpc/RequestHook.re]: Remove this function

File: scan/src/rpc/RequestHook.re [hunk 3]
```diff
@@ -67,3 +95,28 @@ let getRequest = reqID => {
   let json = AxiosHooks.use({j|zoracle/request/$reqID|j});
   json |> Belt.Option.map(_, Request.decode);
 };
+
+// TODO: mock for now
+let getRequestList = (~page=1, ~limit=10, ()) => {
```
Comment (taobun) [inline scan/src/rpc/RequestHook.re]: getList

File: scan/src/rpc/RequestHook.re [hunk 4]
```diff
@@ -67,3 +95,28 @@ let getRequest = reqID => {
   let json = AxiosHooks.use({j|zoracle/request/$reqID|j});
```
Comment (taobun) [inline scan/src/rpc/RequestHook.re]: Change to id

**Review by taobun — CHANGES_REQUESTED** [review]

File: scan/src/rpc/RequestHook.re [hunk 5]
```diff
@@ -17,53 +17,120 @@ module Report = {
...
+      rawDataRequests: json |> field("rawDataRequests", list(RawDataRequest.decode)),
+      reports: json |> field("reports", list(Report.decode)),
+      result:
+        json
+        |> optional(at(["result", "data"], string))
+        |> Belt.Option.map(_, JsBuffer.fromBase64),
     };
 
-  let decode = json => JsonUtils.Decode.(json |> field("result", decodeResult));
+  let decode = json => JsonUtils.Decode.(json |> field("result", decode));
```
Comment (taobun) [inline scan/src/rpc/RequestHook.re]: Rename this function

**Review by taobun — CHANGES_REQUESTED** [review]

**Comment by taobun** [comment]
Do we need to fix in DataSourceHook that refers to this Request type?

**Review by sorawit — DISMISSED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**